restlicsharpclient/restliclient/transport/TransportClient.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using restlicsharpclient.restliclient.response;
+using restlicsharpclient.restliclient.request;
+
+namespace restlicsharpclient.restliclient.transport
+{
+    public interface TransportClient
+    {
+        HttpResponse RestRequestAsync(HttpRequest request);
```

Comment:
This signature may need to be changed later, you can comment out this method for now until you are clear about the signature.